fix(docs-infra): do not include GitHub links in Table of Content

gkalpak opened the pull request:
The docs template for cli commands ([cli-command.template.html][1]) includes an `h2` element with GitHub links for [long description][2]. Since the content of `h2` elements is replicated in the auto-generated Table of Contents, the GitHub links were replicated as well (which is undesirable).

This commit fixes it by explicitly excluding `.github-links` elements, when extracting the content for the ToC (in [TocService#extractHeadingSafeHtml()][3]). This is similar to what we do for the auto-generated `.header-link` anchors.

[1]: https://github.com/angular/angular/blob/1537791f0/aio/tools/transforms/templates/cli/cli-command.template.html
[2]: https://github.com/angular/angular/blob/1537791f0/aio/tools/transforms/templates/cli/cli-command.template.html#L18
[3]: https://github.com/angular/angular/blob/1537791f0/aio/src/app/shared/toc.service.ts#L56

petebacondarwin commented on scripts/git/commit-msg.js:
```diff
@@ -24,7 +24,10 @@ if (msgFile) {
+        console.error(
+            '\nCheck CONTRIBUTING.md at the root of the repo for more information.' +
+            '\n' +
+            '\n(In case you need the invalid commit message, it should be stored in \'.git/COMMIT_EDITMSG\'.)');
```
LOL did you get bitten by this?

gkalpak replied:
More times than I want to admit 😇

(I thought I had removed this change from this PR. I submitted #32420.)

mary-poppins commented:
You can preview 129dee6 at https://pr32418-129dee6.ngbuilds.io/.

petebacondarwin reviewed (changes requested):
Actually there seem to be a bunch of legitimate failures.

mary-poppins commented:
You can preview cab10e0 at https://pr32418-cab10e0.ngbuilds.io/.

mary-poppins commented:
You can preview 48fabd5 at https://pr32418-48fabd5.ngbuilds.io/.

gkalpak committed:
fix(docs-infra): exclude heading anchor icon text from ToC tooltips

The Table of Contents (ToC) is auto-generated based on the content of
heading elements on the page. At the same time, anchor links are
auto-generated and added to each heading element. Note that the Material
Icons used for the anchor icon make use of ligatures, which means that
the icons are specified by using their textual name as text content of
the icon element. As a result, the name of the icon is included in the
parent element's `textContent`.

Previously, the `TocService` used to strip off these anchor elements
when generating the content of ToC items, but not when generating the
content of their tooltips. Thus, tooltips for ToC items would
confusingly include a `link` suffix (`link` is the textual name of the
icon used in heading anchor links).

This commit fixes this by deriving the tooltip content from the
transformed text content (which already has anchor links stripped off),
instead of from the original heading content.

gkalpak committed:
fix(docs-infra): do not include GitHub links in Table of Content

The docs template for cli commands ([cli-command.template.html][1])
includes an `h2` element with GitHub links for [long description][2].
Since the content of `h2` elements is replicated in the auto-generated
Table of Contents, the GitHub links were replicated as well (which is
undesirable).

This commit fixes it by explicitly excluding `.github-links` elements,
when extracting the content for the ToC (in
[TocService#extractHeadingSafeHtml()][3]). This is similar to what we do
for the auto-generated `.header-link` anchors.

[1]: https://github.com/angular/angular/blob/1537791f0/aio/tools/transforms/templates/cli/cli-command.template.html
[2]: https://github.com/angular/angular/blob/1537791f0/aio/tools/transforms/templates/cli/cli-command.template.html#L18
[3]: https://github.com/angular/angular/blob/1537791f0/aio/src/app/shared/toc.service.ts#L56

mary-poppins commented:
You can preview dea05af at https://pr32418-dea05af.ngbuilds.io/.

gkalpak commented:
It turns out there was another issue with tooltips (`title` attribute) of ToC items that the tests didn't hit before. I added another commit to fix tha too.

@petebacondarwin: PTAL 👀

mary-poppins commented:
You can preview 0dd6d5a at https://pr32418-0dd6d5a.ngbuilds.io/.
